src/pytest_bdd/parser.py

```diff
@@ -334,9 +337,15 @@ def name(self) -> str:
         lines = [self._name] + [multilines_content]
         return "\n".join(lines).strip()
 
+    @property
+    def name(self) -> str:
+        return self.full_name
+
     @name.setter
     def name(self, value: str) -> None:
         self._name = value
+        if "full_name" in self.__dict__:
+            del self.__dict__["full_name"]
```

Comment:
```suggestion
        if "full_name" in self.__dict__:
            del self.full_name
```

Reply:
I did try that first but it broke all the tests.

From https://stackoverflow.com/questions/59899732/python-cached-property-how-to-delete

"Your code will try to delete the method itself, and not the cached value. The correct way to do it is to delete the key from self.__dict__ (which is where the data is actually stored)."

Reply:
fixed it